<launch>
    <arg name="robot_version" default="$(optenv ROBOT_VERSION 40)"/>
    <arg name="visualize" default="false"/>
    <arg name="eef_z_bias" default="-0.17"/> <!-- bias added to the z-coordinate of the end-effector -->
    <arg name="model_path" default="$(find kuavo_assets)/models/biped_s$(arg robot_version)/urdf/drake/biped_v3_arm.urdf"/>
    <arg name="model_path_vis" default="$(find kuavo_assets)/models/biped_s$(arg robot_version)/urdf/biped_s$(arg robot_version).urdf"/>
    <arg name="control_hand_side" default="2"/> <!-- 0: left, 1: right, 2: both -->

    <param name="eef_z_bias" value="$(arg eef_z_bias)"/>
    <param name="model_path" value="$(arg model_path)"/>
    <param name="robot_version"    value="$(arg robot_version)"/>
    <param name="control_hand_side" value="$(arg control_hand_side)"/>

    <node name="arms_ik_node" pkg="motion_capture_ik" type="arms_ik_node" output="screen"/>

    <!-- visualize the robot in rviz -->
    <group if="$(arg visualize)">    
        <node name="visulize_traj_node" pkg="motion_capture_ik" type="visulize_traj.py" output="screen"/>
        <arg name="model" />
        <param name="robot_description" textfile="$(arg model_path_vis)" />
        <!-- <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" /> -->
        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find motion_capture_ik)/rviz/mc_robot_state.rviz" />    
    </group>

</launch>
